docs-template/explanation/charm-architecture.md

```diff
@@ -0,0 +1,98 @@
+# Charm architecture
+
+Add overview material here:
+1) What kind of application is it? What kind of software does it use?
+2) Describe Pebble services.
+3) Include an architecture diagram.
+
+<!-- Example: Indico
+At its core, [Indico](https://getindico.io/) is a [Flask](https://flask.palletsprojects.com/) application that integrates with [PostgreSQL](https://www.postgresql.org/), [Redis](https://redis.io/), and [Celery](https://docs.celeryq.dev/en/stable/).
+
+The charm design leverages the [sidecar](https://kubernetes.io/blog/2015/06/the-distributed-system-toolkit-patterns/#example-1-sidecar-containers) pattern to allow multiple containers in each pod with [Pebble](https://juju.is/docs/sdk/pebble) running as the workload container’s entrypoint.
+
+Pebble is a lightweight, API-driven process supervisor that is responsible for configuring processes to run in a container and controlling those processes throughout the workload lifecycle.
+
+Pebble `services` are configured through [layers](https://github.com/canonical/pebble#layer-specification), and the following containers represent each one a layer forming the effective Pebble configuration, or `plan`:
+
+1. An [NGINX](https://www.nginx.com/) container, which can be used to efficiently serve static resources, as well as be the incoming point for all web traffic to the pod.
+2. The [Indico](https://getindico.io/) container itself, which has a [uWSGI](https://uwsgi-docs.readthedocs.io/en/latest/) server configured in HTTP mode.
+
+
+As a result, if you run a `kubectl get pods` on a namespace named for the Juju model you've deployed the Indico charm into, you'll see something like the following:
+
+```bash
+NAME                             READY   STATUS    RESTARTS   AGE
+indico-0                         3/3     Running   0         6h4m
+```
+
+This shows there are 4 containers - the three named above, as well as a container for the charm code itself.
+
+And if you run `kubectl describe pod indico-0`, all the containers will have as Command ```/charm/bin/pebble```. That's because Pebble is responsible for the processes startup as explained above.
+-->
+
+## OCI images
+
+We use [Rockcraft](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/) to build OCI Images for <charm-name>. 
+The images are defined in [<charm-name> rock](link to rock).
+They are published to [Charmhub](https://charmhub.io/), the official repository of charms.
+
+> See more: [How to publish your charm on Charmhub](https://juju.is/docs/sdk/publishing)
+
+## Containers
+
+Configuration files for the containers can be found in the respective directories that define the rocks.
+
+<!--
+### Container example
+
+Description of container.
+
+The workload that this container is running is defined in the [<container-name> rock](link to rock).
+-->
+
+## Metrics
+Inside the above mentioned containers, additional Pebble layers are defined in order to provide metrics.
```

Comment:
For machine charms: Metrics can also apply to non-containerised workloads (e.g. in the github-runner charm its the runner-manager), but it could also be a snap or something else.

Reply:
Thanks for the information, I'll revise the text